scripts/cert-manager/app-registration-cert-mananger.sh

```diff
@@ -0,0 +1,308 @@
+
+    # Create new .json file with updated credential.
+    UPDATED_SECRET_VALUES_FILE="updated_secret_values.json"
+    test -f "$UPDATED_SECRET_VALUES_FILE" && rm "$UPDATED_SECRET_VALUES_FILE"
+
+    cat <<EOF >>${UPDATED_SECRET_VALUES_FILE}
+{
+"name": "${APP_REGISTRATION_CERT_MANAGER}",
+"id": "${APP_ID}",
+"password": "${APP_PASSWORD}",
+"description": "${APP_DESCRIPTION}",
+"tenantId": "${APP_TENANT}"
+}
+EOF
+
+    # Update keyvault with secret
+    printf "Updating keyvault \"$AZ_RESOURCE_KEYVAULT\"..."
+    if [[ $(az keyvault secret set --name "$APP_REGISTRATION_CERT_MANAGER" --vault-name "$AZ_RESOURCE_KEYVAULT" --file "$UPDATED_SECRET_VALUES_FILE" --expires "$EXPIRATION_DATE" 2>&1) == *"ERROR"* ]]; then
+        az keyvault secret set --name "$APP_REGISTRATION_CERT_MANAGER" --vault-name "$AZ_RESOURCE_KEYVAULT" --file "$UPDATED_SECRET_VALUES_FILE" --expires "$EXPIRATION_DATE"
+        echo -e "\nERROR: Could not update secret in keyvault \"$AZ_RESOURCE_KEYVAULT\". Exiting..."
+        exit 1
+    fi
+    printf " Done\n"
+
+    rm $UPDATED_SECRET_VALUES_FILE
+fi
+
+#######################################################################################
+### Assign custom permission on TXT records
+###
+
+ROLENAME="DNS TXT Contributor"
+APP_ID=$(az role definition list --name "$ROLENAME" -o tsv)
+CRED_ROLE=true
+echo ""
+if [[ $APP_ID == "" ]]; then
+    echo "Role \"$ROLENAME\" does not exist."
+
+    if [[ $USER_PROMPT == true ]]; then
+        while true; do
+            read -p "Create Role $ROLENAME? (Y/n) " yn
+            case $yn in
+                [Yy]* ) break;;
+                [Nn]* ) CRED_ROLE=false; echo "Skipping."; break;;
+                * ) echo "Please answer yes or no.";;
+            esac
+        done
+    fi
+
+    printf "Creating app registration..."
+
+    if [[ $CRED_ROLE == true ]]; then
+        CUSTOMDNSROLE_JSON="cert-mananger-custom-dns-role.json"
+        test -f "$CUSTOMDNSROLE_JSON" && rm "$CUSTOMDNSROLE_JSON"
+    cat <<EOF >>${CUSTOMDNSROLE_JSON}
+    {
+        "Name": "$ROLENAME",
+        "Id": "",
+        "IsCustom": true,
+        "Description": "Can manage DNS TXT records only.",
+        "Actions": [
+            "Microsoft.Network/dnsZones/TXT/*",
+            "Microsoft.Network/dnsZones/read",
+            "Microsoft.Authorization/*/read",
+            "Microsoft.Insights/alertRules/*",
+            "Microsoft.ResourceHealth/availabilityStatuses/read",
+            "Microsoft.Resources/deployments/*",
+            "Microsoft.Resources/subscriptions/resourceGroups/read",
+            "Microsoft.Support/*"
+        ],
+        "NotActions": [
+        ],
+        "AssignableScopes": [
+            "/subscriptions/${AZ_SUBSCRIPTION_ID}/resourceGroups/${AZ_RESOURCE_GROUP_COMMON}/providers/Microsoft.Network/dnszones/${AZ_RESOURCE_DNS}"
+        ]
+    }
+EOF
+
+
+    az role definition create --role-definition ./cert-mananger-custom-dns-role.json
```

Comment:
Use the variable for the filename. Also indentation
```suggestion
        az role definition create --role-definition "$CUSTOMDNSROLE_JSON"
```